peerconnection.go

```diff
@@ -1079,15 +1171,119 @@ func (pc *PeerConnection) drainSRTP() {
 				return
 			}
 
-			_, ssrc, err := srtpSession.AcceptStream()
+			r, ssrc, err := srtpSession.AcceptStream()
 			if err != nil {
 				pc.log.Warnf("Failed to accept RTP %v", err)
 				return
 			}
 
-			if !handleUndeclaredSSRC(ssrc) {
-				pc.log.Warnf("Incoming unhandled RTP ssrc(%d), OnTrack will not be fired", ssrc)
-			}
+			pc.mu.Lock()
+			pc.pendingReadStreamsSRTP[ssrc] = r
+			pc.mu.Unlock()
+
+			go func() {
+				// test first N (10) packets, if no mid and rid are found give up
+				c := 0
+
+				// read incoming packet until we can populate mid and rid from packet
+				// extensions (not all packet will provide such information) so continue reading until we'll find both
+				var mid, rid string
+				for {
+					b := make([]byte, receiveMTU)
```

Comment:
This and the `rtp.Packet` below should be created outside the loop and reused (you may have to clear a value or two our of the rtp packet before unmarshalling each time...the lib should be using a buf pool IMO, but that's a different topic)